lib/phraseapp_updater/locale_file/loader.rb

```diff
@@ -0,0 +1,24 @@
+require 'phraseapp_updater/locale_file'
+
+class PhraseAppUpdater
+  class LocaleFile
+    class Loader
+      def initialize(extension)
+        @extension = extension
+      end
+
+      def load(filename)
+        unless File.readable?(filename) && File.file?(filename)
+          raise RuntimeError.new("Couldn't read localization file at #{filename}")
+        end
+
+        LocaleFile.class_for_file_format(@extension).new(File.basename(filename).chomp(".#{@extension}"), File.read(filename))
```

Comment:
I'll probably not worry about that one too much, but that is a good point about using this pattern. It's mostly meant to catch the path being wrong and print a friendly message, though.